src/vs/workbench/parts/files/browser/views/openEditorsViewer.ts

```diff
@@ -216,6 +216,21 @@ export class Controller extends treedefaults.DefaultController {
 		super({ clickBehavior: treedefaults.ClickBehavior.ON_MOUSE_DOWN });
 	}
 
+	public onClick(tree: ITree, element: any, event: IMouseEvent): boolean {
+
+		// Close opened editor on middle mouse click
+		if (element instanceof OpenEditor && event.browserEvent && event.browserEvent.button === 1 /* Middle Button */) {
+			const closeAction = this.instantiationService.createInstance(CloseEditorAction, CloseEditorAction.ID, CloseEditorAction.LABEL);
+			closeAction.run({ event, group: (<OpenEditor>element).editorGroup, editor: (<OpenEditor>element).editorInput }).done(() => {
+				closeAction.dispose();
+			}, errors.onUnexpectedError);
+
+			return true;
+		}
+
+		return super.onClick(tree, element, event);
+	}
+
 	protected onLeftClick(tree: ITree, element: any, event: IMouseEvent, origin: string = 'mouse'): boolean {
```

Comment:
Looks good - tried it out and it works!
Though why do you instantiate the action when you can simply go to the editorService and say closeEditor - you have the position and the input.